src/cls/rgw/cls_rgw.cc

```diff
@@ -2110,6 +2110,13 @@ int rgw_dir_suggest_changes(cls_method_context_t hctx,
         return -EINVAL;
       }
 
+      if (cur_disk.pending_map.size() == 0) {
```

Comment:
note that there's already a `if (cur_disk.pending_map.empty()) {` condition below that prevents dir suggestions from being applied against pending entries

Reply:
ok, i understand this PR better after the discussions around https://github.com/ceph/ceph/pull/45345

this early return means that we'll only apply suggestions for entries that *had* pending operations but they expired by tag timeout. the change in https://github.com/ceph/ceph/pull/45345 only addresses a race between dir_suggest and the complete, but this change *also* prevents clients from making suggestions against completed entries

in bucket listing, we have [several conditions](https://github.com/ceph/ceph/blob/87c4cf0/src/rgw/rgw_rados.cc#L8667-L8674) for calling `check_disk_state()` - `!pending_map.empty()` is not the only one. is it possible that we're relying on dir_suggest to work even though pending_map is empty?